exporters/common/src/main/java/io/opentelemetry/exporter/internal/TlsUtil.java

```diff
@@ -77,14 +76,11 @@ public static X509KeyManager keyManager(byte[] privateKeyPem, byte[] certificate
       PrivateKey key = generatePrivateKey(keySpec, SUPPORTED_KEY_FACTORIES);
 
       CertificateFactory cf = CertificateFactory.getInstance("X.509");
-
-      List<Certificate> chain = new ArrayList<>();
-
       ByteArrayInputStream is = new ByteArrayInputStream(certificatePem);
-      while (is.available() > 0) {
-        chain.add(cf.generateCertificate(is));
-      }
-
+      // pass the input stream to generateCertificates to get a list of certificates
+      // generateCertificates can handle multiple certificates in a single input stream
+      // including PEM files with explanatory text
+      List<? extends Certificate> chain = (List<? extends Certificate>) cf.generateCertificates(is);
```

Comment:
I wrote a test case for this, but was surprised that it passes on the `main` branch. Not sure why it would pass if this was required for handle explanatory text:
```
  @TempDir private Path tempDir;

  private static final String EXPLANATORY_TEXT =
      "Subject: CN=Foo\n"
          + "Issuer: CN=Foo\n"
          + "Validity: from 7/9/2012 3:10:38 AM UTC to 7/9/2013 3:10:37 AM UTC\n";

  /**
   * Append <a href="https://datatracker.ietf.org/doc/html/rfc7468#section-5.2">explanatory text</a>
   * prefix and verify {@link TlsUtil#keyManager(byte[], byte[])} succeeds.
   */
  @ParameterizedTest
  @MethodSource("keyManagerArgs")
  void keyManager_CertWithExplanatoryText(SelfSignedCertificate selfSignedCertificate)
      throws IOException {
    Path certificate = tempDir.resolve("certificate");
    Files.write(certificate, EXPLANATORY_TEXT.getBytes(StandardCharsets.UTF_8));
    Files.write(
        certificate,
        com.google.common.io.Files.toByteArray(selfSignedCertificate.certificate()),
        StandardOpenOption.APPEND);

    assertThatCode(
        () ->
            TlsUtil.keyManager(
                com.google.common.io.Files.toByteArray(selfSignedCertificate.privateKey()),
                com.google.common.io.Files.toByteArray(new File(certificate.toString()))))
        .doesNotThrowAnyException();
  }

  private static Stream<Arguments> keyManagerArgs() throws CertificateException {
    Instant now = Instant.now();
    return Stream.of(
        Arguments.of(
            new SelfSignedCertificate(Date.from(now), Date.from(now), "RSA", 2048),
            new SelfSignedCertificate(Date.from(now), Date.from(now), "EC", 256)));
  }
```

Reply:
I definitely faced this problem with our internal certificate bundle, but after your comment, I debugged a little more and I am wondering if the bundle is not created correctly. 
When I iterated using our internal bundle with the main branch code, it added 12 certificates correctly ( even with the explanatory text ) but at the end there was one byte which ended up throwing "empty input" error.

So main branch code is able to handle explanatory text but `cf.generateCertificates(is)` is a little more forgiving. 
I completely understand if you prefer to not merge this PR.

Reply:
I'm happy enough with switching to `cf.generateCertificates(is)` if its a more robust parsing solution, but just need to demonstrate that with some test case. I.e. some certificate that fails with the logic on `main` but succeeds when this logic is applied.